src/components/Menu/MenuItem.tsx

```
@@ -144,6 +146,7 @@ class MenuItem extends AutoControlledComponent<ReactProps<MenuItemProps>, MenuIt
     defaultMenuOpen: PropTypes.bool,
     onActiveChanged: PropTypes.func,
     inSubmenu: PropTypes.bool,
+    kind: PropTypes.string,
```

```suggestion
    kind: PropTypes.oneOf(['divider']),
```

---

Why should the type here be `oneOf` containing divider? MenuItem will actually never be created if the kind prop is divider. For the menu item it can be 'item' according to the convention, or nothing as this is the default component. Am I wrong assuming this?

---

So, `kind: PropTypes.oneOf(['divider', 'item']),`? We should disallow to pass invalid values to `kind`

---

My point was that in the MenuItem, it should be `oneOf(['item'])`, and in the divider should be `oneOf(['divider'])` as the MenuItem will never be created if the kind prop is something other then item (or null, but the prop is anyways optional - anything else provided should throw an error), and the MenuDivider will never be created if the kind prop is something other then the 'divider'. Would this be sufficient?

---

Resolved in personal conversation :)